src/eko/io/items.py

```diff
@@ -0,0 +1,204 @@
+        return cls(**asdict(matching), inverse=inverse)
+
+    @property
+    def as_atlas(self) -> matchings.Matching:
+        """The associated segment."""
+        return matchings.Matching(self.scale, self.hq)
+
+
+Recipe = Union[Evolution, Matching]
+
+
+@dataclass(frozen=True)
+class Target(Header):
+    """Target evolution point, labeling evolution from origin to there."""
+
+    scale: SquaredScale
+    nf: FlavorsNumber
+
+    @classmethod
+    def from_ep(cls, ep: EPoint):
+        """Create instance from the :class:`EPoint` analogue."""
+        return cls(*ep)
+
+    @property
+    def ep(self) -> EPoint:
+        """Cast to :class:`EPoint`."""
+        return (self.scale, self.nf)
+
+
+@dataclass(frozen=True)
+class Operator:
+    """Operator representation.
+
+    To be used to hold the result of a computed 4-dim operator (either a raw
+    evolution operator or a matching condition).
+
+    Note
+    ----
+    IO works with streams in memory, in order to avoid intermediate write on
+    disk (keep read from and write to tar file only).
+
+    """
+
+    operator: npt.NDArray
+    """Content of the evolution operator."""
+    error: Optional[npt.NDArray] = None
+    """Errors on individual operator elements (mainly used for integration
+    error, but it can host any kind of error).
+    """
+
+    def save(self, stream: BinaryIO) -> bool:
+        """Save content of operator to bytes.
+
+        The content is saved on a `stream`, in order to be able to perform the
+        operation both on disk and in memory.
+
+        The returned value tells whether the operator saved contained or not
+        the error (this control even the format, ``npz`` with errors, ``npy``
+        otherwise).
+
+        """
+        aux = io.BytesIO()
+        if self.error is None:
+            np.save(aux, self.operator)
+        else:
+            np.savez(aux, operator=self.operator, error=self.error)
+        aux.seek(0)
+
+        # compress if requested
+        content = lz4.frame.compress(aux.read())
+
+        # write compressed data
+        stream.write(content)
+        stream.seek(0)
+
+        # return the type of array dumped (i.e. 'npy' or 'npz')
+        return self.error is None
+
+    @classmethod
+    def load(cls, stream: BinaryIO):
+        """Load operator from bytes.
+
+        An input `stream` is used to load the operator from, in order to
+        support the operation both on disk and in memory.
+
+        """
+        extracted_stream = io.BytesIO(lz4.frame.decompress(stream.read()))
+        content = np.load(extracted_stream)
+
+        if isinstance(content, np.ndarray):
+            op = content
+            err = None
+        elif isinstance(content, npyio.NpzFile):
+            op = content["operator"]
+            err = content["error"]
+        else:
+            # TODO: We might consider dropping this exception since np.load will always
```

Comment:
The other option is to drop the one of the two `if`s conditions, and replace them with a plain `else`.
However, I find it more readable this way, since the user knows what's happening just reading the code, without the need of any further tool.

Reply:
Fine - but this triggers a completely  fake test: https://github.com/NNPDF/eko/blob/ef1289d8abbc8f9e7daafab28e8bca050343226c/tests/eko/io/test_items.py#L77

Reply:
Only together with the full-coverage-at-all-costs requirement :)

Reply:
fine ... recall that [I just *did* find another bug](https://github.com/NNPDF/eko/commit/b94e4a21bdc97422505a323fc491ffdefdba51a7) doing this and it was exactly in the one line missing ...